core.lua

```diff
@@ -225,10 +274,65 @@ local ACE_OPTIONS = {
             set = function(info, val) VocalCooldowns.db.profile.update_period_seconds = val end,
             get = function(info) return VocalCooldowns.db.profile.update_period_seconds end
         },
+        ["headsup_seconds"] = {
+            order = 20,
+            name = "Headsup (seconds)",
+            desc = "Notify this many seconds before the cooldown is available.",
+            width = "double",
+            type = "range",
+            min = 0.1,
+            max = 5,
+            step = 0.1,
+            set = function(info, val) VocalCooldowns.db.profile.headsup_seconds = val end,
+            get = function(info) return VocalCooldowns.db.profile.headsup_seconds end
+        },
+        ["voice_selection"]  = {
+            type = "select",
+            name = "Select TTS Voice",
+            desc = "Choose a voice for TTS.",
+            values = voiceOptions,
+            get = function(info) return VocalCooldowns.db.profile.selected_voice end,
+            set = function(info, value)
+                VocalCooldowns.db.profile.selected_voice = value
+                Speech("Selected new voice "..voiceOptions[value])
+            end,
+        },
+        ["notification_template"] = {
+            type = "input",
+            name = "Notification Template",
+            desc = "Enter the template for TTS notifications.",
+            get = function(info) return VocalCooldowns.db.profile.notification_template or "" end,
+            set = function(info, value)
+                VocalCooldowns.db.profile.notification_template = value
+            end,
+            multiline = false,  -- Set to true if you want a larger text box for longer messages
+            width = "full",     -- Expands the input box to the full width of the config dialog
+        },
+        ["add_blacklist_item"] = {
+            type = "input",
+            name = "Add Item",
+            desc = "Enter a string to add to the list.",
+            get = function(info) return "" end,  -- Always return an empty string so the field resets after input
+            set = function(info, value)
+                table.insert(VocalCooldowns.db.profile.blacklist, value)
+                tprint(VocalCooldowns.db.profile.blacklist,2)
```

Comment:
That seems to be for debugging, should be marked to avoid unnecessary chat messages.

Reply:
Yes, I have a cleaned up version I'm using to play myself, but I also removed a bunch of your debugging code and added some logic to avoid concurrency problems I observed in a raid setting. If you want I can push it here for review, but the changes are quite extensive.